cmd/SplitGrantsGovXMLDB/handler_test.go

```diff
@@ -602,54 +599,46 @@ func TestProcessRecord(t *testing.T) {
 
 	t.Run("Error getting item from DynamoDB", func(t *testing.T) {
 		setupLambdaEnvForTesting(t)
-		c := mockS3ReadwriteObjectAPI{
-			mockHeadObjectAPI(
-				func(ctx context.Context, params *s3.HeadObjectInput, optFns ...func(*s3.Options)) (*s3.HeadObjectOutput, error) {
-					t.Helper()
-					return &s3.HeadObjectOutput{}, fmt.Errorf("server error")
-				},
-			),
-			mockGetObjectAPI(nil),
-			mockPutObjectAPI(nil),
-		}
+		s3client := mockPutObjectAPI(func(context.Context, *s3.PutObjectInput, ...func(*s3.Options)) (*s3.PutObjectOutput, error) {
+			t.Helper()
+			require.Fail(t, "PutObject called unexpectedly")
+			return nil, nil
+		})
 		ddbLookups := make(mockDDBClientGetItemCollection, 0)
 		ddbLookups = append(ddbLookups, mockDDBClientGetItemReturnValue{
 			GrantId:          string(testOpportunity.OpportunityID),
 			ItemLastModified: string(testOpportunity.LastUpdatedDate),
 			GetItemErr:       errors.New("Some issue with DynamoDB"),
 		})
-		err := processRecord(context.TODO(), c, ddbLookups.NewGetItemClient(t), testOpportunity)
+		err := processRecord(context.TODO(), s3client, ddbLookups.NewGetItemClient(t), testOpportunity)
```

Comment:
This change corresponds to the simplified `PutObjectAPI` argument in the `processRecord()` signature.